docs/contributor/adr/0011-rate-limit.md

```diff
@@ -0,0 +1,476 @@
+| local.buckets.tokensPerFill       | int                 | The number of tokens added to the bucket during each fill interval.                                                                                                                                                                                                                                                                                                                                | yes      |
+| local.buckets.fillInterval        | duration            | The fill interval that tokens are added to the bucket. <br/>During each fill interval, `tokensPerFill` are added to the bucket. The bucket will never contain more than `maxTokens` tokens. The `fillInterval` must be greater than or equal to 50ms to avoid excessive refills.                                                                                                                   | yes      |
+| enableResponseHeaders             | boolean             | Enables **x-rate-limit** response headers. The default value is `false`.                                                                                                                                                                                                                                                                                                                           | no       |
+| enforce                           | boolean             | Specifies whether the rate limit should be enforced. The default value is `true`.                                                                                                                                                                                                                                                                                                                            | no       |
+
+### Usage Example
+
+```yaml
+apiVersion: operator.kyma-project.io/v1alpha1
+kind: RateLimit
+metadata:
+  name: httpbin-local-rate-limit
+  namespace: default
+spec:
+  selectorLabels:
+    app: httpbin
+  local:
+    defaultBucket:
+      maxTokens: 10
+      tokensPerFill: 5
+      fillInterval: 30s
+    buckets:
+      - path: /headers
+        headers:
+          x-api-version: v1
+        bucket:
+          maxTokens: 2
+          tokensPerFill: 2
+          fillInterval: 30s
+      - path: /ip
+        bucket:
+          maxTokens: 20
+          tokensPerFill: 10
+          fillInterval: 30s
+  enableResponseHeaders: true
+```
+
+The following diagram illustrates Istio Controller technical design extended with the RateLimit controller loop.
```

Comment:
I think this is not up to date and also the linked image does not exist in this repository.

Reply:
Let's remove it here for now. It will be updated soon in the technical design part